jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -285,6 +289,16 @@ public HiveConnection(String uri, Properties info) throws SQLException {
     this(uri, info, HiveJdbcBrowserClientFactory.get());
   }
 
+  public HiveConnection Clone() throws SQLException {
```

Comment:
In which use case the Clone is needed? It is not better just to create another connection with the same connection string?

Reply:
It ensures both clients share a session ID.

Reply:
I'm not sure overriding Object#clone makes sense. I've added a comment to describe it's behavior.

Reply:
if you don't won't to override Object#clone, create a copy constructor

Reply:
switched to copy constructor